precompiles/utils/src/solidity_conversions.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright 2019-2021 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+use sp_core::U256;
+use sp_std::vec::Vec;
+
+// Solidity's bool type is 256 bits as shown by these examples
+// https://docs.soliditylang.org/en/v0.8.0/abi-spec.html
+// This utility function converts a Rust bool into the corresponding Solidity type
+pub fn bool_to_solidity_bytes(b: bool) -> Vec<u8> {
```

Comment:
Perhaps this could be a `const fn` and could be something like:

```rust
match b {
    true => 1usize.into(),
    false => 0usize.into(),
}
```

Reply:
And then convert the U256 to vec?

Reply:
Maybe. Isn't that the purpose of the following fn?

Reply:
Yeah perhaps is more clear that way

Reply:
having constant functions does not allow to do function calls or conversions, so I guess I will leave it as it is for now